### ISSUE TITLE
Crash in read_BIN2R() when reading an empty file with forcedVersion

### ISSUE THREAD

**Issue by mcol** [R-Lum/Luminescence#225]
See this example:
```R
zero <- tempfile(pattern = "zero", fileext = ".binx")
write(raw(), zero)

## this is fine
read_BIN2R(zero, verbose = FALSE)
# Error: [read_BIN2R()] BIN/BINX format version (0a) is not supported or file is broken. Supported
#  version numbers are: 03, 04, 05, 06, 07, 08

## this crashes
read_BIN2R(zero, verbose = FALSE, forced.VersionNumber = 8)
# Error in if (temp.RECTYPE != 0 & temp.RECTYPE != 1 & temp.RECTYPE != 128) { : 
#  argument is of length zero
```
When we force the version number, the code keeps reading the input file for the next few fields, but it never checks whether they are valid, so it eventually crashes.

**Comment by RLumSK** [comment]
Thanks, this is really an edge case but it makes sense to check this.

**Comment by mcol** [comment]
Fixed by #226.
